cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -662,6 +663,84 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
             return Future.succeededFuture(this);
         }
 
+        Future<ReconciliationState> zkVersionChange() {
+
+            String kafkaSsName = KafkaCluster.kafkaClusterName(name);
+            String zkSsName = ZookeeperCluster.zookeeperClusterName(name);
+
+            Future<StatefulSet> kafkaSsFuture = kafkaSetOperations.getAsync(namespace, kafkaSsName);
+            Future<StatefulSet> zkSsFuture = zkSetOperations.getAsync(namespace, zkSsName);
+
+            return CompositeFuture.all(kafkaSsFuture, zkSsFuture).compose(allFutures -> {
+                if (allFutures.succeeded()) {
+
+                    StatefulSet zkSs = zkSsFuture.result();
+                    StatefulSet kafkaSs = kafkaSsFuture.result();
+
+                    if (zkSs == null || kafkaSs == null) {
+                        return Future.succeededFuture(this);
+                    }
+
+                    KafkaVersionChange versionChange = getKafkaVersionChange(kafkaSs);
+
+                    log.debug("Checking if we need to change the Zookeeper image for Zookeeper cluster {}", zkSsName);
+
+                    if (versionChange.isNoop()) {
+                        log.debug("Kafka.spec.kafka.version is unchanged therefore no change to Zookeeper is required");
+                        return Future.succeededFuture(this);
+                    } else {
+                        if (versionChange.isDowngrade()) {
+                            log.debug("Kafka downgrade from {} to {} requires Zookeeper downgrade from {} to {}",
```

Comment:
This looks like it will be printed when Kafka version downgrades to another version using the same Zoo version. Or?

Reply:
Good point, I added a method to `KafkaVersionChange` to check if the zk version changes and simplified the if statement here.